consensus/spos/worker.go

```diff
@@ -243,6 +243,7 @@ func (wrk *Worker) ProcessReceivedMessage(message p2p.MessageP2P) error {
 	if wrk.consensusService.IsMessageWithBlockHeader(msgType) {
 		headerHash := cnsDta.BlockHeaderHash
 		header := wrk.blockProcessor.DecodeBlockHeader(cnsDta.SubRoundData)
+		//TODO: Could be added here an additional check for prev random seed vs random seed and proposer validity
```

Comment:
rephrase the TODO.

Reply:
This is a must, to add all the checks everywhere.

Reply:
Yes, missing also in interceptors. Should be a new component as it will gather some code.

Reply:
Rephrased